apps/judicial-system/backend/src/app/modules/file/file.service.ts

```diff
@@ -207,15 +207,18 @@ export class FileService {
       )
     }
 
+    const fileName = createFile.key.slice(
+      isIndictmentCase(theCase.type)
+        ? INDICTMENT_NAME_BEGINS_INDEX
+        : NAME_BEGINS_INDEX,
+    )
+
     return this.fileModel.create({
       ...createFile,
       state: CaseFileState.STORED_IN_RVG,
       caseId: theCase.id,
-      name: createFile.key.slice(
-        isIndictmentCase(theCase.type)
-          ? INDICTMENT_NAME_BEGINS_INDEX
-          : NAME_BEGINS_INDEX,
-      ),
+      name: fileName,
+      userGeneratedFilename: fileName.replace(/\.pdf$/, ''),
```

Comment:
Not an ideal solution, but I feel like it's the cleanest at this time and solves the issue at hand. In the future maybe we could rename the field since this isn't technically user generated when created. Just rename it display name or something like that.